src/main/java/uk/gov/pay/products/resources/ProductResource.java

```diff
@@ -183,4 +197,15 @@ public Response findProductsAndStats(@QueryParam("gatewayAccountId") Integer gat
         List<ProductUsageStat> usageStats = productFactory.productFinder().findProductsAndUsage(gatewayAccountId);
         return Response.status(OK).entity(usageStats).build();
     }
+
+    @POST
+    @Path("/products/{productExternalId}/regenerate-api-token")
+    @Produces(APPLICATION_JSON)
+    @Consumes(APPLICATION_JSON)
+    public Response regenerateProductApiToken(@PathParam("productExternalId") String productExternalId) {
+        return productFactory.productFinder().findByExternalId(productExternalId).map((product) -> {
```

Comment:
Don’t really need the brackets around `(product)`.

Reply:
Will remove the brackets from a single parameter next time.